scripts/generator/graphql_generator/csharp/SDK/iOS/iOSWebCheckout.cs.erb

```diff
@@ -36,12 +51,42 @@ namespace <%= namespace %>.SDK.iOS {
             if (message.Content == "loadedThankYou") {
                 // Do a round trip to the server to see if the purchase is solid.
                 // Invoke callbacks on cart to pass to Unity
-                OnSuccess();
+                DidCheckoutCompleteSuccessfully(CurrentCheckout.id(), (didComplete) => {
+                    if (didComplete) {
+                        OnSuccess();
+                    } else {
+                        var error = new ShopifyError(
+                            ShopifyError.ErrorType.CheckoutError, 
+                            "User navigated to end of checkout without successfully checking out."
+                        );
+
+                        OnFailure(error);
+                    }
+                });
             } else if (message.Content == "cancelled") {
                 OnCancelled();
                 message.Respond("");
             }
         }
+
+        private delegate void CheckoutCheckCallback(bool didComplete);
+        private void DidCheckoutCompleteSuccessfully(string checkoutId, CheckoutCheckCallback callback) {
+            var query = new QueryRootQuery();
+            query.node(
+                buildQuery: n => n
+                    .onCheckout(c => c.completedAt()),
+                id: CurrentCheckout.id()
+            );
+            
+            Client.Query(query, (response, error) => {
+                var checkout = (Checkout) response.node();
+                if (error != null) {
+                    OnFailure(error);
```

Comment:
I'm not sure we should be calling `OnFailure` here. As a function `DidCheckoutCompleteSuccessfully` should return if checkout completed successfully or not via the `callback`. If it didn't complete successfully then you might want to attach the error to the callback. 

Not calling `callback` here puts more burden on the implementor when they wonder why their callback was never called, on success or failure.